qiskit/opflow/__init__.py

```diff
@@ -17,6 +17,12 @@
 
 .. currentmodule:: qiskit.opflow
 
+.. deprecated:: 0.24.0
+
+    The :mod::`qiskit.opflow` module is deprecated and will be removed no earlier
+    than 3 months after the release date. For code migration guidelines,
+    visit https://qisk.it/opflow_migration.
+
```

Comment:
Adding a deprecation warning at the top should be enough, right? I considered reiterating this for each submodule listed.

Reply:
This only causes the deprecation to show up when reading docs. It does not issue a runtime warning. For that, you need to call `warnings.warn()` because I did not yet add a helper function for deprecating a module.

Reply:
Right, that is why I kept the warning in this page in particular. Once you go to the class docs, you should get the specific deprecation warning for that class.